vnext/Microsoft.ReactNative/Views/VirtualTextViewManager.cpp

```diff
@@ -106,7 +140,13 @@ bool VirtualTextViewManager::UpdateProperty(
   // FUTURE: In the future cppwinrt will generate code where static methods on
   // base types can be called.  For now we specify the base type explicitly
   if (TryUpdateForeground<winrt::TextElement>(span, propertyName, propertyValue)) {
-    static_cast<VirtualTextShadowNode *>(nodeToUpdate)->m_highlightData.foregroundColor = ColorFrom(propertyValue);
+    auto node = static_cast<VirtualTextShadowNode *>(nodeToUpdate);
+    if (IsValidOptionalColorValue(propertyValue)) {
+      node->m_foregroundColor = OptionalColorFrom(propertyValue);
+      const auto propertyChangeType =
+          node->m_foregroundColor ? PropertyChangeType::AddBackgroundColor : PropertyChangeType::None;
```

Comment:
We could potentially split the background color and foreground color props. If a foreground color is set but nothing else has a foreground color, we don't need to recurse through the tree.

Also, since we need this for foreground color as well as background color, we should generalize the PropertyChangeType to call it "Highlight" or something.

Reply:
This still isn't working quite right. Thankfully the E2E tests identify the problem. Will fix.

Reply:
Need to also set m_hasDescendantBackgroundColor on node as well.